From 96e11d35f2130d5cbc14669637252f79555e85f6 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Tue, 19 Aug 2014 19:30:53 -0400
Subject: [scsi] qla4xxx: Fix smatch warning in func qla4xxx_get_ep_param

Message-id: <1408476658-5056-16-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88354
O-Subject: [PATCH RHEL 7.1 e-stor 15/20] qla4xxx: Fix smatch warning in func qla4xxx_get_ep_param
Bugzilla: 1089349
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Adheer Chandravanshi <adheer.chandravanshi@qlogic.com>

scsi commit id 73597bdd8326b2c8a68cd046fd2ac338982cebf1

Fix following smatch warning:-
 drivers/scsi/qla4xxx/ql4_os.c:1752 qla4xxx_get_ep_param() warn: variable dereferenced before check 'qla_ep' (see line 1745)

Signed-off-by: Vikas Chaudhary <vikas.chaudhary@qlogic.com>
Reviewed-by: Mike Christie <michaelc@cs.wisc.edu>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla4xxx/ql4_os.c b/drivers/scsi/qla4xxx/ql4_os.c
index 6446956..8508eb3 100644
--- a/drivers/scsi/qla4xxx/ql4_os.c
+++ b/drivers/scsi/qla4xxx/ql4_os.c
@@ -1742,6 +1742,9 @@ static int qla4xxx_get_ep_param(struct iscsi_endpoint *ep,
  struct sockaddr *dst_addr;
  struct scsi_qla_host *ha;
 
+ if (!qla_ep)
+  return -ENOTCONN;
+
  ha = to_qla_host(qla_ep->host);
  DEBUG2(ql4_printk(KERN_INFO, ha, "%s: host: %ld\n", __func__,
      ha->host_no));
@@ -1749,9 +1752,6 @@ static int qla4xxx_get_ep_param(struct iscsi_endpoint *ep,
  switch (param) {
  case ISCSI_PARAM_CONN_PORT:
  case ISCSI_PARAM_CONN_ADDRESS:
-  if (!qla_ep)
-   return -ENOTCONN;
-
   dst_addr = (struct sockaddr *)&qla_ep->dst_addr;
   if (!dst_addr)
    return -ENOTCONN;
-- 
1.7.1